enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,989 @@
+
+## Design Details
+
+### Open Questions
+
+None.
+
+### Test Plan
+
+We should have at least tests that verifies that the upgrade can be performed in
+a fully disconnected environment, both for a single node cluster and a cluster
+with multiple nodes. These tests should gate the OCP release.
+
+It is desirable to have another test that scans the OCP components looking for
+use of the `Always` pull policy. This should probably run for each pull request
+of each OCP component, and prevent merging if it detects that the offending pull
+policy is used. We should consider adding admission in CI for this.
+
+### Graduation Criteria
+
+The feature will ideally be introduced as `Dev Preview` in OpenShift 4.14.z,
+moved to `Tech Preview` in 4.15 and declared `GA` in 4.16.
+
+#### Dev Preview -> Tech Preview
+
+- Ability to upgrade a single-node clusters using the `bundle.enable: false` mode
+(no bundle, just pinning and pre-loading of the images).
+
+- Availability of the tests that verify the upgrade if single-node clusters.
+
+- Availability of the tests that verify that no OCP component uses the `Always`
+pull policy.
+
+- Obtain positive feedback from at least one customer.
+
+#### Tech Preview -> GA
+
+- Ability to upgrade single-node and multi-node clusters using the
+`bundle.enable` set to `true` or `false`: with a bundle or just pinning and
+pre-loading the images.
+
+- Ability to create bundles using the `oc adm upgrade create bundle` command.
+
+- Availability of the tests that verify the upgrade in single-node and
+multi-node clusters.
+
+- User facing documentation created in
+[https://github.com/openshift/openshift-docs](openshift-docs).
+
+#### Removing a deprecated feature
+
+Not applicable, no feature will be removed.
+
+### Upgrade / Downgrade Strategy
+
+There are no additional considerations for upgrade or downgrade. The same
+considerations that apply to the cluster version operator in general will also
+apply in this case.
+
+### Version Skew Strategy
+
+This feature will only be usable once the cluster version operator and the
+machine config operator have been upgraded to support it. That upgrade will have
+to be done by other means.
+
+For subsequent upgrades we will ensure that the cluster version operator can
+work with both the old and the new version of the machine config operator.
+
+### Operational Aspects of API Extensions
+
+Not applicable, there are no API extensions.
+
+#### Failure Modes
+
+#### Support Procedures
+
+## Implementation History
+
+There is an initial prototype exploring some of the implementation details
+described here in this [https://github.com/jhernand/upgrade-tool](repository).
+
+## Alternatives
+
+The alternative to this is to make a registry server available, either outside
+or inside the cluster.
+
+An external registry server is a well known solution, even for disconnected
+environments, but it is not feasible in most of the target environments.
```

Comment:
Why not? Where did this statement come from?

Reply:
Added a link to the motivation section.